src/ui-day.c

```diff
@@ -39,44 +39,35 @@
 struct day_item day_cut[38] = { {0, 0, 0, {NULL}} };
 
 /*
- * Return the selected item in the APP panel.
+ * Return the selected APP item.
+ * This is a pointer into the day vector and invalid after a day vector rebuild,
+ * but the (order, item) data may be used to refind the object.
  */
-struct day_item *ui_day_selitem(void)
+struct day_item *ui_day_get_sel(void)
 {
 	if (day_item_count(0) <= 0)
-		return NULL;
+		return &empty_day;
 
 	return day_get_item(listbox_get_sel(&lb_apt));
 }
 
 /*
- * Return the day (midnight) of the selected item in the APP panel.
+ * Set the selected item from the saved day_item.
  */
-time_t ui_day_selday(void)
+void ui_day_find_sel(void)
 {
-	return update_time_in_date(ui_day_selitem()->order, 0, 0);
-}
-
-/*
- * Return true if the selected APP item is the dummy event.
- */
-int ui_day_dummy(void)
-{
-	struct day_item *d = ui_day_selitem();
+	int n;
 
-	return d ? d->start == DUMMY : 0;
-}
-
-void ui_day_set_selitem_by_aptev_ptr(union aptev_ptr p)
-{
-	int n = day_get_position_by_aptev_ptr(p);
-	if (n >= 0)
+	if ((n = day_sel_index()) != -1);
```

Comment:
This looks suspicious. Are you sure you want that semicolon?

Reply:
A genuine bug. Corrected in next upload.